TestWorkflowTemplateRefWithShutdownAndSuspend flacky

## Summary 

What happened/what you expected to happen?

## Diagnostics

What Kubernetes provider are you using? 

What version of Argo Workflows are you running? 

```yaml
Paste a workflow that reproduces the bug, including status:
kubectl get wf -o yaml ${workflow} 
```

```
Paste the logs from the workflow controller:
kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name) | grep ${workflow}
```

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

This is still happening after https://github.com/argoproj/argo-workflows/pull/5381 got merged. An example CI run here: https://github.com/argoproj/argo-workflows/pull/5372/checks?check_run_id=2122737311

```
 --- FAIL: TestWorkflowTemplateRefWithShutdownAndSuspend (0.12s)
    --- FAIL: TestWorkflowTemplateRefWithShutdownAndSuspend/WorkflowTemplateRefWithShutdownStop (0.00s)
        operator_workflow_template_ref_test.go:356: 
            	Error Trace:	operator_workflow_template_ref_test.go:356
            	Error:      	"" does not contain "workflow shutdown with strategy"
            	Test:       	TestWorkflowTemplateRefWithShutdownAndSuspend/WorkflowTemplateRefWithShutdownStop
        operator_workflow_template_ref_test.go:357: 
            	Error Trace:	operator_workflow_template_ref_test.go:357
            	Error:      	"" does not contain "Stop"
            	Test:       	TestWorkflowTemplateRefWithShutdownAndSuspend/WorkflowTemplateRefWithShutdownStop
```

--- a/workflow/controller/controller_test.go
+++ b/workflow/controller/controller_test.go
@@ -473,7 +473,7 @@ const wfWithTmplRef = `
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
-  generateName: workflow-template-hello-world-
+  name: workflow-template-hello-world
   namespace: default
 spec:
   entrypoint: whalesay-template

